GCC Compile errors when using full optimizations

**Describe the bug**
When compiling with full optimizations enabled, gcc detects and reports some additional errors that are not seen when compiling in debug mode.

For example:
```
/home/joe/code/cfecfs/github/libs/bplib/mpool/src/v7_mpool.c: In function ‘bplib_mpool_search_list’:
/home/joe/code/cfecfs/github/libs/bplib/mpool/src/v7_mpool.c:841:16: error: ‘iter.position’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  841 |     return iter.position;
      |            ~~~~^~~~~~~~~
cc1: all warnings being treated as errors
```

**To Reproduce**
Compile with `CMAKE_BUILD_TYPE` set to Release.

**Expected behavior**
Should build and run cleanly

**System observed on:**
Ubuntu 22.04 (gcc 11.3)

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/app/bpcat.c
+++ b/app/bpcat.c
@@ -108,9 +108,13 @@ bp_handle_t storage_intf_id;
 static void app_quick_exit(int signo)
 {
     static const char message[] = "Caught CTRL+C\n";
+    ssize_t           sz;
 
     signal(signo, SIG_DFL);
-    write(STDERR_FILENO, message, sizeof(message) - 1);
+
+    /* squench warning about unused result - no recourse on failure to write to STDERR */
+    sz = write(STDERR_FILENO, message, sizeof(message) - 1);
+    (void)sz;
     app_running = 0;
 }
 
@@ -689,6 +693,7 @@ static void *app_out_entry(void *arg)
 
     desc                = arg;
     curr_bp_buffer      = NULL;
+    curr_app_buffer     = NULL;
     curr_stream_pos     = 0;
     total_message_count = 0;
     segment_remain      = 0;

--- a/mpool/src/v7_mpool.c
+++ b/mpool/src/v7_mpool.c
@@ -825,6 +825,8 @@ bplib_mpool_block_t *bplib_mpool_search_list(const bplib_mpool_block_t *list, bp
     int                     status;
     bplib_mpool_list_iter_t iter;
 
+    memset(&iter, 0, sizeof(iter));
+
     status = bplib_mpool_list_iter_goto_first(list, &iter);
     while (status == BP_SUCCESS)
     {

--- a/store/file_offload.c
+++ b/store/file_offload.c
@@ -210,6 +210,7 @@ static void bplib_file_offload_sid_to_name(bplib_file_offload_state_t *state, ch
 static int bplib_file_offload_write_block_content(int fd, bplib_file_offload_record_t *rec, const void *ptr, size_t sz)
 {
     ssize_t res;
+    int     status;
 
     if (ptr == NULL)
     {
@@ -218,15 +219,25 @@ static int bplib_file_offload_write_block_content(int fd, bplib_file_offload_rec
 
     rec->num_bytes += sz;
     rec->crc = bplib_crc_update(&BPLIB_CRC32_CASTAGNOLI, rec->crc, ptr, sz);
-    res      = write(fd, ptr, sz);
-    assert(res == sz);
 
-    return BP_SUCCESS;
+    res = write(fd, ptr, sz);
+    if (res == sz)
+    {
+        status = BP_SUCCESS;
+    }
+    else
+    {
+        bplog(NULL, BP_FLAG_DIAGNOSTIC, "write(): %s\n", strerror(errno));
+        status = BP_ERROR;
+    }
+
+    return status;
 }
 
 static int bplib_file_offload_read_block_content(int fd, bplib_file_offload_record_t *rec, void *ptr, size_t sz)
 {
     ssize_t res;
+    int     status;
 
     if (ptr == NULL)
     {
@@ -239,11 +250,20 @@ static int bplib_file_offload_read_block_content(int fd, bplib_file_offload_reco
     }
 
     rec->num_bytes -= sz;
-    res = read(fd, ptr, sz);
-    assert(res == sz);
 
-    rec->crc = bplib_crc_update(&BPLIB_CRC32_CASTAGNOLI, rec->crc, ptr, sz);
-    return BP_SUCCESS;
+    res = read(fd, ptr, sz);
+    if (res == sz)
+    {
+        status   = BP_SUCCESS;
+        rec->crc = bplib_crc_update(&BPLIB_CRC32_CASTAGNOLI, rec->crc, ptr, sz);
+    }
+    else
+    {
+        bplog(NULL, BP_FLAG_DIAGNOSTIC, "write(): %s\n", strerror(errno));
+        status = BP_ERROR;
+    }
+
+    return status;
 }
 
 static int bplib_file_offload_write_payload(int fd, bplib_file_offload_record_t *rec,
@@ -497,8 +517,12 @@ static int bplib_file_offload_offload(bplib_mpool_block_t *svc, bp_sid_t *sid, b
         memset(&rec, 0, sizeof(rec));
         rec.check_val = BPLIB_FILE_OFFLOAD_MAGIC;
         rec.crc       = bplib_crc_initial_value(&BPLIB_CRC32_CASTAGNOLI);
-        off           = lseek(fd, sizeof(rec), SEEK_SET);
-        assert(off == sizeof(rec));
+
+        off = lseek(fd, sizeof(rec), SEEK_SET);
+        if (off != sizeof(rec))
+        {
+            bplog(NULL, BP_FLAG_DIAGNOSTIC, "lseek(): %s\n", strerror(errno));
+        }
 
         result = bplib_file_offload_write_blocks(fd, &rec, pblk);
         if (result == BP_SUCCESS)
@@ -511,9 +535,9 @@ static int bplib_file_offload_offload(bplib_mpool_block_t *svc, bp_sid_t *sid, b
             {
                 result = BP_ERROR;
             }
-
-            close(fd);
         }
+
+        close(fd);
     }
 
     return result;
@@ -539,6 +563,7 @@ static int bplib_file_offload_restore(bplib_mpool_block_t *svc, bp_sid_t sid, bp
     pool = bplib_mpool_get_parent_pool_from_link(svc);
 
     result = BP_ERROR;
+    pblk   = NULL;
 
     bplib_file_offload_sid_to_name(state, bundle_file, sizeof(bundle_file), sid, false);
 

